cni/pkg/taint/taint_test.go

```diff
@@ -0,0 +1,286 @@
+			},
+			args: args{
+				config: istiocniConfig,
+			},
+			wants: map[string][]string{
+				"istio-cni": {"istio-cni", "kube-system", "app=istio"},
+			},
+		},
+		{
+			name: "general config",
+			fields: fields{
+				client: fakeClientset([]v1.Pod{}, []v1.Node{}, []v1.ConfigMap{istiocniConfig}),
+			},
+			args: args{
+				config: combinedConfig,
+			},
+			wants: map[string][]string{
+				"istio-cni": {"istio-cni", "kube-system", "app=istio"},
+				"others":    {"others", "blah", "app=others"},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := TaintSetter{
+				Client: fake.NewSimpleClientset(),
+			}
+			ts.LoadConfig(tt.args.config)
+			for _, elem := range ts.configs {
+				if tt.wants[elem.Name] == nil {
+					t.Errorf("wants to load = %v", elem.Name)
+				}
+				if tt.wants[elem.Name][0] != elem.Name {
+					t.Errorf("wants to load name = %v found %v", elem.Name, tt.wants[elem.Name][0])
+				}
+				if tt.wants[elem.Name][1] != elem.Namespace {
+					t.Errorf("wants to load namespace = %v found %v", elem.Namespace, tt.wants[elem.Name][1])
+				}
+				if tt.wants[elem.Name][2] != elem.LabelSelector {
+					t.Errorf("wants to load selector = %v found %v", elem.LabelSelector, tt.wants[elem.Name][2])
+				}
+			}
+		})
+	}
+}
+
+func TestTaintSetter_AddReadinessTaint(t *testing.T) {
+	type fields struct {
+		client kubernetes.Interface
+	}
+	type args struct {
+		node v1.Node
+	}
+	tests := []struct {
+		name     string
+		fields   fields
+		args     args
+		wantList []v1.Taint
+	}{
+		{
+			name: "working node already get taint",
+			fields: fields{
+				client: fakeClientset([]v1.Pod{workingPod}, []v1.Node{testingNode}, []v1.ConfigMap{}),
+			},
+			args: args{
+				testingNode,
+			},
+			wantList: []v1.Taint{{Key: TaintName, Effect: v1.TaintEffectNoSchedule}},
+		},
+		{
+			name: "plain node add readiness taint",
+			fields: fields{
+				client: fakeClientset([]v1.Pod{workingPod}, []v1.Node{plainNode}, []v1.ConfigMap{}),
+			},
+			args: args{
+				plainNode,
+			},
+			wantList: []v1.Taint{{Key: TaintName, Effect: v1.TaintEffectNoSchedule}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := TaintSetter{
+				Client: tt.fields.client,
+			}
+			err := ts.AddReadinessTaint(&tt.args.node)
+			if err != nil {
+				t.Errorf("error happened in readiness %v", err.Error())
+				return
+			}
+			updatedNode, err := ts.Client.CoreV1().Nodes().Get(context.TODO(), tt.args.node.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Errorf("error happened in readiness %v", err.Error())
+				return
+			}
+			if !reflect.DeepEqual(updatedNode.Spec.Taints, tt.wantList) {
+				t.Errorf("AddReadinessTaint() gotList = %v, want %v", updatedNode.Spec.Taints, tt.wantList)
+			}
+		})
+	}
+}
+func TestTaintSetter_HasReadinessTaint(t *testing.T) {
+	type fields struct {
+		client kubernetes.Interface
+	}
+	type args struct {
```

Comment:
nit: if args has 1 param, no need to have a struct. just inline it. Same applies to `fields` and other tests

Reply:
changed all args and fields in unit tests